.github/workflows/ci.yml

```diff
@@ -52,6 +69,63 @@ jobs:
         if: matrix.os == 'ubuntu-latest' && matrix.node == 14
         uses: codecov/codecov-action@v3
 
+  types:
+    needs: prepare
+    name: Types (${{ matrix.os}}, Node ${{ matrix.node }}, TS ${{ matrix.typescript }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        node: [14]
+        typescript: ['4.5', '4.6', '4.7', '4.8', '4.9']
+
+    steps:
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Retrieve node_modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ matrix.os }}-node-v${{ matrix.node }}-deps-${{ hashFiles(format('{0}{1}', github.workspace, '/package-lock.json')) }}
+
+      - name: Overwrite TypeScript
+        run: npm install --save-dev typescript@${{ matrix.typescript }}
```

Comment:
Is the `--save-dev` necessary? Modifying the `package-lock.json` makes the workflow less pure since it modifies the code its testing.

We could use `--no-save` to only install TypeScript without modifying the repo.

(If you think changing `package-lock.json` doesn't make a difference, we can keep it as is.)

Reply:
Brilliant! Yes